tasks/cycles/cycles.js

```diff
@@ -106,3 +105,99 @@ for (let elem in personIncomes) {
 }
 console.log(sum);
 
+
+
+function reverseString(str) {
+   return str;
+}
+
+console.log(reverseString('hello')); // olleh
+
+function reverseString(str) {
+  return str;
+}
+
+
+function reverseString(str) {
+  var newString = "";
+  for (var i = str.length - 1; i >= 0; i--) {
+    newString += str[i];
+  }
+  return newString;
+}
+console.log(reverseString("hello"));
+
+function reverseString(phreak) {
+  var daFunkaFuck = "";
+  for (var i = phreak.length - 1; i >= 0; i--) {
+    daFunkaFuck += phreak[i]; 
+  }
+  return daFunkaFuck;
+}
+console.log(reverseString('!akcoR feihC a ma I'));
+
+//2. Write a method to find an index of minimal item from an array;
+function findMin(phreak) {
+  var daFunkaFuck = "";
+  for (var i = phreak.length - 1; i >= 0; i--) {
+    daFunkaFuck <= phreak[i];
+  }
+  return daFunkaFuck;
+}
+console.log(findMin);
+
+console.log(chars.indexOf(name));
+
+function showProps(obj, objName) {
+  var result = "";
+  for (var i in obj) {
+    if (obj.hasOwnProperty(i)) {
+      result += objName + "." + i + " = " + obj[i] + "\n";
+    }
+  }
+  return result;
+}
+
+var square = function (number) 
+{ return number + " " + number; };
+var x = square("Jim");
+console.log(x);
+
+var factorial = function fac(n) { return n < 2 ? 1 : n * fac(n - 1); };
+
+console.log(factorial(10));
+
+var dummyWalks;
+for (dummyWalks = 0; dummyWalks < 19; dummyWalks++)
+{
+ console.log("Dummy walks along... bitch") 
+}
+
+function findBiggusDickus(hate, crop) {
```

Comment:
WTF is this?

Reply:
oh.. practice excerpt sorry. will remove.